### ISSUE TITLE
Flaky test: GoAwayThenClose

### ISSUE THREAD

**Issue by dfawley** [grpc/grpc-go#3253]
From the most recent cron run, although I've seen it at least one other time this week:
```
--- FAIL: Test (159.44s)
    --- FAIL: Test/GoAwayThenClose (20.00s)
        end2end_test.go:7160: UnaryCall(_) = _, rpc error: code = DeadlineExceeded desc = context deadline exceeded; want _, nil
FAIL
```

**Comment by dfawley** [comment]
Able to reproduce locally after about 10k attempts.

**Comment by dfawley** [comment]
Log output!
```
I1213 09:22:11.251132   46065 clientconn.go:245] parsed scheme: "bz4gqd3ubu3h"
I1213 09:22:11.251200   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:42297  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:11.251478   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:11.251514   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:11.251877   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:42297  <nil> 0 <nil>} {127.0.0.1:39297  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:11.251906   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:42297  <nil> 0 <nil>}, addrs: [{127.0.0.1:42297  <nil> 0 <nil>} {127.0.0.1:39297  <nil> 0 <nil>}]
I1213 09:22:11.251914   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:11.252142   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:42297  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:42297: connect: connection refused". Reconnecting...
I1213 09:22:31.251254   46065 types.go:69] attempt to delete child with id 6 from a parent (id=3) that doesn't currently exist
I1213 09:22:31.251274   46065 types.go:69] attempt to delete child with id 6 from a parent (id=3) that doesn't currently exist
--- FAIL: Test (20.01s)
    --- FAIL: Test/GoAwayThenClose (20.00s)
        end2end_test.go:7160: UnaryCall(_) = _, rpc error: code = DeadlineExceeded desc = context deadline exceeded; want _, nil
I1213 09:22:31.258265   46065 clientconn.go:245] parsed scheme: "bz4gqmaq1tdu"
I1213 09:22:31.258301   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:39085  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.258335   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.258402   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.259061   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:39085  <nil> 0 <nil>} {127.0.0.1:37073  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.259114   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:39085  <nil> 0 <nil>}, addrs: [{127.0.0.1:39085  <nil> 0 <nil>} {127.0.0.1:37073  <nil> 0 <nil>}]
I1213 09:22:31.259133   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.259480   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:39085  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:39085: connect: connection refused". Reconnecting...
I1213 09:22:31.264767   46065 types.go:69] attempt to delete child with id 16 from a parent (id=14) that doesn't currently exist
I1213 09:22:31.264786   46065 types.go:69] attempt to delete child with id 16 from a parent (id=14) that doesn't currently exist
I1213 09:22:31.270935   46065 clientconn.go:245] parsed scheme: "bz4gqmaxl4gs"
I1213 09:22:31.270981   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:36287  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.271025   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.271112   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.271631   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:36287  <nil> 0 <nil>} {127.0.0.1:38067  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.271684   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:36287  <nil> 0 <nil>}, addrs: [{127.0.0.1:36287  <nil> 0 <nil>} {127.0.0.1:38067  <nil> 0 <nil>}]
I1213 09:22:31.271702   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.272071   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:36287  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:36287: connect: connection refused". Reconnecting...
I1213 09:22:31.274837   46065 types.go:69] attempt to delete child with id 26 from a parent (id=24) that doesn't currently exist
I1213 09:22:31.274853   46065 types.go:69] attempt to delete child with id 26 from a parent (id=24) that doesn't currently exist
I1213 09:22:31.279291   46065 clientconn.go:245] parsed scheme: "bz4gqmb2kwl0"
I1213 09:22:31.279306   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:45107  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.279318   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.279358   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.279762   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:45107  <nil> 0 <nil>} {127.0.0.1:46123  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.279812   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:45107  <nil> 0 <nil>}, addrs: [{127.0.0.1:45107  <nil> 0 <nil>} {127.0.0.1:46123  <nil> 0 <nil>}]
I1213 09:22:31.279831   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.280186   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:45107  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:45107: connect: connection refused". Reconnecting...
I1213 09:22:31.284249   46065 types.go:69] attempt to delete child with id 36 from a parent (id=34) that doesn't currently exist
I1213 09:22:31.284264   46065 types.go:69] attempt to delete child with id 36 from a parent (id=34) that doesn't currently exist
I1213 09:22:31.288141   46065 clientconn.go:245] parsed scheme: "bz4gqmb7uf4z"
I1213 09:22:31.288162   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:37855  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.288181   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.288227   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.288574   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:37855  <nil> 0 <nil>} {127.0.0.1:35287  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.288629   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:37855  <nil> 0 <nil>}, addrs: [{127.0.0.1:37855  <nil> 0 <nil>} {127.0.0.1:35287  <nil> 0 <nil>}]
I1213 09:22:31.288645   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.288939   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:37855  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:37855: connect: connection refused". Reconnecting...
I1213 09:22:31.291380   46065 types.go:69] attempt to delete child with id 46 from a parent (id=44) that doesn't currently exist
I1213 09:22:31.291402   46065 types.go:69] attempt to delete child with id 46 from a parent (id=44) that doesn't currently exist
I1213 09:22:31.295114   46065 clientconn.go:245] parsed scheme: "bz4gqmbbzxcc"
I1213 09:22:31.295134   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:37711  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.295151   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.295184   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.295545   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:37711  <nil> 0 <nil>} {127.0.0.1:33811  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.295590   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:37711  <nil> 0 <nil>}, addrs: [{127.0.0.1:37711  <nil> 0 <nil>} {127.0.0.1:33811  <nil> 0 <nil>}]
I1213 09:22:31.295608   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.295894   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:37711  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:37711: connect: connection refused". Reconnecting...
I1213 09:22:31.299251   46065 types.go:69] attempt to delete child with id 55 from a parent (id=54) that doesn't currently exist
I1213 09:22:31.299268   46065 types.go:69] attempt to delete child with id 55 from a parent (id=54) that doesn't currently exist
I1213 09:22:31.303704   46065 clientconn.go:245] parsed scheme: "bz4gqmbh3q2q"
I1213 09:22:31.303739   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:45825  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.303778   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.303851   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.304293   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:45825  <nil> 0 <nil>} {127.0.0.1:34101  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.304358   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:45825  <nil> 0 <nil>}, addrs: [{127.0.0.1:45825  <nil> 0 <nil>} {127.0.0.1:34101  <nil> 0 <nil>}]
I1213 09:22:31.304385   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.304806   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:45825  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:45825: connect: connection refused". Reconnecting...
I1213 09:22:31.307724   46065 types.go:69] attempt to delete child with id 66 from a parent (id=64) that doesn't currently exist
I1213 09:22:31.307739   46065 types.go:69] attempt to delete child with id 66 from a parent (id=64) that doesn't currently exist
I1213 09:22:31.313047   46065 clientconn.go:245] parsed scheme: "bz4gqmbmoaj9"
I1213 09:22:31.313071   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:37135  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.313095   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.313138   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.313518   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:37135  <nil> 0 <nil>} {127.0.0.1:35205  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.313578   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:37135  <nil> 0 <nil>}, addrs: [{127.0.0.1:37135  <nil> 0 <nil>} {127.0.0.1:35205  <nil> 0 <nil>}]
I1213 09:22:31.313601   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.313967   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:37135  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:37135: connect: connection refused". Reconnecting...
I1213 09:22:31.317136   46065 types.go:69] attempt to delete child with id 76 from a parent (id=74) that doesn't currently exist
I1213 09:22:31.317160   46065 types.go:69] attempt to delete child with id 76 from a parent (id=74) that doesn't currently exist
I1213 09:22:31.320665   46065 clientconn.go:245] parsed scheme: "bz4gqmbr7g38"
I1213 09:22:31.320690   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:43621  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.320728   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.320780   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.321121   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:43621  <nil> 0 <nil>} {127.0.0.1:33041  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.321151   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:43621  <nil> 0 <nil>}, addrs: [{127.0.0.1:43621  <nil> 0 <nil>} {127.0.0.1:33041  <nil> 0 <nil>}]
I1213 09:22:31.321162   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.321503   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:43621  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:43621: connect: connection refused". Reconnecting...
I1213 09:22:31.324185   46065 types.go:69] attempt to delete child with id 86 from a parent (id=84) that doesn't currently exist
I1213 09:22:31.324197   46065 types.go:69] attempt to delete child with id 86 from a parent (id=84) that doesn't currently exist
I1213 09:22:31.328930   46065 clientconn.go:245] parsed scheme: "bz4gqmbw4lao"
I1213 09:22:31.328955   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:46025  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.328977   46065 clientconn.go:659] ClientConn switching balancer to "pick_first"
I1213 09:22:31.329027   46065 picker_wrapper.go:213] blockingPicker: the picked transport is not ready, loop back to repick
I1213 09:22:31.329391   46065 resolver_conn_wrapper.go:178] ccResolverWrapper: sending update to cc: {[{127.0.0.1:46025  <nil> 0 <nil>} {127.0.0.1:45433  <nil> 0 <nil>}] <nil> <nil>}
I1213 09:22:31.329443   46065 clientconn.go:822] addrConn: tryUpdateAddrs curAddr: {127.0.0.1:46025  <nil> 0 <nil>}, addrs: [{127.0.0.1:46025  <nil> 0 <nil>} {127.0.0.1:45433  <nil> 0 <nil>}]
I1213 09:22:31.329457   46065 clientconn.go:842] addrConn: tryUpdateAddrs curAddrFound: true
W1213 09:22:31.329729   46065 clientconn.go:1288] grpc: addrConn.createTransport failed to connect to {127.0.0.1:46025  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:46025: connect: connection refused". Reconnecting...
I1213 09:22:31.333619   46065 types.go:69] attempt to delete child with id 96 from a parent (id=94) that doesn't currently exist
I1213 09:22:31.333641   46065 types.go:69] attempt to delete child with id 96 from a parent (id=94) that doesn't currently exist
FAIL
FAIL	google.golang.org/grpc/test	20.173s
FAIL
```

**Comment by menghanl** [comment]
This flaky was "fixed" once in #2822 (PR #2855). But seems it didn't fix all the flakes?
